core/mo-services-impl/nmf-platform-generic-impl/src/main/java/esa/mo/platform/impl/provider/gen/GPSProviderServiceImpl.java

```diff
@@ -776,13 +850,64 @@ public void getTIMEASentence(GetTIMEASentenceInteraction interaction)
     }
     interaction.sendAcknowledgement();
     try {
-      String resp = adapter.getTIMEASentence();
-      interaction.sendResponse(resp);
+      interaction.sendResponse(adapter.getTIMEASentence());
     } catch (IOException e) {
       interaction
           .sendError(new MALStandardError(PlatformHelper.DEVICE_NOT_AVAILABLE_ERROR_NUMBER, null));
       e.printStackTrace();
     }
   }
 
+  /**
+   * Checks the fixQquality on the GNSS position.
+   * @return true if a fixed position has been established, false otherwise
+   */
+  private boolean isPositionFixed() {
+    boolean isFixed = false;
+    try {
+      if(adapter.getCurrentPosition().getExtraDetails().getFixQuality() > 0){
+        isFixed = true;
+      }
+    } catch (NullPointerException e)
+    {
+      LOGGER.warning("Could not receive a fixed position: " + e.getMessage());
+    }
+    return isFixed;
+  }
+
+  private Position getTLEPropagatedPosition()
+  {
+    Calendar targetDate = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+    SpacecraftState state = getSpacecraftState(targetDate);
+
+
+    // Converting to geodetic lat/lon/alt
+    Frame ecf = FramesFactory.getITRF(IERSConventions.IERS_2010,true);
+    OneAxisEllipsoid earth = new OneAxisEllipsoid(Constants.WGS84_EARTH_EQUATORIAL_RADIUS, Constants.WGS84_EARTH_FLATTENING, ecf);
+    GeodeticPoint satLatLonAlt = earth.transform(state.getPVCoordinates().getPosition(), FramesFactory.getEME2000(),
+            state.getDate());
+
+    PositionExtraDetails extraDetails = new PositionExtraDetails(new Time(targetDate.getTimeInMillis()),
+            0, 0, 0.0f,0.0f, PositionSourceType.TLE);
+
+    return new Position((float) satLatLonAlt.getLatitude(), (float) satLatLonAlt.getLongitude(),
+            (float) satLatLonAlt.getAltitude(), extraDetails);
+  }
+
+  private SpacecraftState getSpacecraftState(Calendar targetDate) {
+
+    //setup orekit
+    DataProvidersManager manager = DataProvidersManager.getInstance();
+    manager.addProvider(OrekitResources.getOrekitData());
```

Comment:
Wouldn't adding it on every call resource-heavy, or leaking memory?

Reply:
Right, forgot to polish this part. Will make it to load it just once.